specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -0,0 +1,127 @@
+# AWS Lambda instrumentation
+This spec specifies the instrumentation of applications / services running on AWS Lambda.
+
+An AWS Lambda application needs to implement a **handler method** that is called whenever that Lambda function is invoked. A handler method receives (at least) two objects with useful information:
+- `event`: Depending on the trigger type of the Lambda function this object contains different, trigger-specific data. For some trigger types this object might be even missing / being a void object.
+- `context`: This object provides generic (trigger agnostic) meta information about the Lambda function.
+
+In our instrumentation we use these objects to derive useful meta and context information.
+
+## Generic Lambda Instrumentation
+In general, to instrument Lambda functions, we create transactions that wrap the execution of that handler method. In cases where we cannot assume any type or information in the `event` object (e.g. if the trigger type is undefined), we ignore trigger-specific information and simply wrap the handler method with a transaction, while using the `context` object to derive some necessary fields.
+
+
+Field | Value | Description 
+---   | ---   | ---
+`transaction.name` | e.g. `MyFunctionName` | The name of the Lambda function. This can be retrieveed either from the `context` object or from the environment variable `AWS_LAMBDA_FUNCTION_NAME`
+`service.framework.name` | `AWS Lambda` | 
+`transaction.type` | `request` |  
+`faas.coldstart` | `true` / `false` | Boolean value indicating whether a Lambda function invocation was a cold start or not.
```

Comment:
Naive Question -- what is the `faas` namespace?  Is this something [we'll be adding to the APM Server transaction spec](https://github.com/elastic/apm-server/blob/master/docs/spec/v2/transaction.json)?  If not where is this data being added?

Reply:
`functions as a service` I think. I'm not sold on the name. Maybe `serverless` or something instead?

Reply:
I took the OTel specification as an orientation for the naming and fields: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/faas.md

We will need to add those fields to the Server Intake.
I also already started an [RFC process](https://github.com/elastic/ecs/pull/1518) to add this block of fields to ECS.